runsc/boot/vfs.go

```diff
@@ -585,6 +585,12 @@ func (c *containerMounter) configureOverlay(ctx context.Context, conf *config.Co
 	// filesystem specific options.
 	upperOpts := *lowerOpts
 	upperOpts.GetFilesystemOptions = vfs.GetFilesystemOptions{InternalMount: true}
+	if conf.GetOverlay2().Size() != "" {
```

Comment:
In gVisor, the overlay can be configured in many scenarios:
- Via --overlay2 flag for rootfs or even bind mounts.
- For specific bind mounts using mount annotations

See https://github.com/google/gvisor/blob/d30c58e5b8a4ead573d82eba3c8c240eafa21af2/runsc/container/container.go#L939

So applying the size limit to every overlay in the sandbox based on --overlay2 flag is incorrect.